Added react app frontend with react-routes

phi-h-nguyen opened the pull request:
Added new React app with typescript and react-router-dom (https://reactrouter.com/docs/en/v6/getting-started/tutorial) for multiple page organization.

McSobkov commented:
I'd like to edit the README before the submission deadline, but otherwise looks great!

McSobkov reviewed (approved):
The readme looks auto-generated, so worth revising in the future, but useful for now.

SeanEZhou commented:
Hi Phi, thanks for setting up the front-end! I have a few suggestions:

- Consider adding typescript as a --save-dev. I had to install typescript manually in order to start the web app, otherwise I get an error about eslint not able to load typescript/plugin. I think adding typescript as --save-dev could solve this.
- Consider changing eslint.config to 4 space indentation to be consistent with other code files.
- Order currently links to the cart page, consider fixing it to properly link to the order page. 

Thanks!

phi-h-nguyen commented:
Made the suggested changes
- added typescript as --save-dev
- fixed order link
- updated project structure

I think we can discuss the indentation style as a team. I personally prefer 2-spaced indentation, especially for for react development with many levels of nesting.

khushi02 reviewed (approved):
This looks good to me! I appreciate the changes you completed from the previous feedback. Following up on Max's comment, I think it is alright to leave the README as it is right now.

SeanEZhou reviewed (approved):
Hi Phi thanks for the changes! Looks good to me now!
